pandas/core/generic.py

```diff
@@ -2909,6 +2909,9 @@ def _drop_axis(self, labels, axis, level=None, errors='raise'):
             else:
                 indexer = ~axis.isin(labels)
 
+            if indexer.all() and errors == 'raise':
```

Comment:
Actually, I might have been a little to quick to say switch to `.all()`.  It looks like builtin `all` will be faster for sizable arrays that have a `False` early on.  Leave it as is for now though, unless someone else says to switch it, since `.all()` should have a lower bound for the worst case scenario.

Switching the `errors == 'raise'` check to be first should be more efficient, since that's a cheaper check, and will cause the `and` to terminate before checking `indexer` in cases where this isn't applicable:

```python
In [2]: a = np.repeat([True], 10**5)

In [3]: errors = 'other'

In [4]: %timeit errors == 'raise' and a.all()
58.7 ns ± 10 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [5]: %timeit a.all() and errors == 'raise'
6.82 µs ± 856 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
```

Reply:
Ok, I'll leave it as is for now and make the other change

Reply:
Looks like you're right about `.all()` when a value is `False` early on:

``` 
In[9]: arr = np.repeat([True], 1000000)
In[10]: arr[15] = False
In[11]: %timeit all(arr)
1.93 µs ± 193 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
In[12]: %timeit arr.all()
244 µs ± 38.4 µs per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```